### ISSUE TITLE
Import performance

### ISSUE THREAD

**Issue by AndreasGerken** [YingfanWang/PaCMAP#40]
Importing pacmap takes ~4.3 seconds on my machine (core I7/ 8th Generation, 40GB RAM, python 3.8.10). Since i want to include it in my package i overtake these 4.3 seconds into my package and delays everything by a lot.

Could you increase the import performance?

Timed with multiple runs of
`python3 -m timeit -r 1 "import pacmap"`

**Comment by hyhuang00** [comment]
I made an investigation into the import process and on my machine it takes even longer to import (~10s). The tuna log doesn't say much about the root cause of the slow import, but I suppose the just in time compiler we used for the functions is the culprit. When you `import pacmap`, the functions will be compiled. I will try to fix this issue by caching some of the compiled function, which hopefully can reduce import latency.
<img width="1678" alt="Screen Shot 2022-07-02 at 3 33 29 PM" src="https://user-images.githubusercontent.com/16739711/177017900-63a09983-76e1-44d4-9482-5ba317a9c669.png">
